go/geth_adapter/adapter_test.go

```diff
@@ -466,3 +471,212 @@ func TestRunContextAdapter_bigIntToWord(t *testing.T) {
 		t.Errorf("Conversion returned wrong value, expected %v, got %v", want, got)
 	}
 }
+
+func TestRunContextAdapter_ConvertRevisionCanDealWithUnsetMergeBlock(t *testing.T) {
+	chainConfig := &params.ChainConfig{
+		ChainID:            big.NewInt(42),
+		MergeNetsplitBlock: nil,
+	}
+
+	revision := convertRevision(chainConfig, big.NewInt(0), 0)
+	if revision != tosca.R07_Istanbul {
+		t.Errorf("Conversion returned wrong value, expected %v, got %v", tosca.R07_Istanbul, revision)
+	}
+}
+
+func TestRunContextAdapter_ConvertRevision(t *testing.T) {
+	cancunTime := uint64(1000)
+	shanghaiTime := uint64(900)
+	parisBlock := big.NewInt(100)
+	londonBlock := big.NewInt(90)
+	berlinBlock := big.NewInt(80)
+	istanbulBlock := big.NewInt(70)
+
+	tests := map[string]struct {
+		mergeBlock *big.Int
+		block      *big.Int
+		time       uint64
+		want       tosca.Revision
+	}{
+		"Istanbul": {
+			block: istanbulBlock,
+			time:  uint64(0),
```

Comment:
params seem to treat nil time as no fork. I think it would be interesting to check that nil is a valid value

Reply:
time is a uint64 in the `ChainConfig` therefore it can not be nil.

Reply:
```go

	chainConfig := &params.ChainConfig{
		ChainID:            big.NewInt(42),
		IstanbulBlock:      istanbulBlock,
		LondonBlock:        londonBlock,
		BerlinBlock:        berlinBlock,
		MergeNetsplitBlock: parisBlock,
		ShanghaiTime:       &shanghaiTime,
		CancunTime:         &cancunTime,
	}
```

I see a pointer

Reply:
But this is the config of the EVM not the input of the function.
This time is nil in other tests such as `TestRunContextAdapter_Run`.